File: go-client-sdk/pkg/models/shared/eventupdate.go
package shared

type EventUpdate struct {
	Additional    map[string]interface{} `json:"additional,omitempty"`
	Description   *string                `json:"description,omitempty"`
	EndDateTime   *string                `json:"endDateTime,omitempty"`
	IsAllDayEvent *bool                  `json:"isAllDayEvent,omitempty"`
	Location      *string                `json:"location,omitempty"`
	OwnerUserID   *string                `json:"ownerUserId,omitempty"`
	StartDateTime *string                `json:"startDateTime,omitempty"`
	Subject       *string                `json:"subject,omitempty"`
	Type          *string                `json:"type,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/eventupdate.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class EventUpdate extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=additional" })
  additional?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=description" })
  description?: string;

  @SpeakeasyMetadata({ data: "json, name=endDateTime" })
  endDateTime?: string;

  @SpeakeasyMetadata({ data: "json, name=isAllDayEvent" })
  isAllDayEvent?: boolean;

  @SpeakeasyMetadata({ data: "json, name=location" })
  location?: string;

  @SpeakeasyMetadata({ data: "json, name=ownerUserId" })
  ownerUserId?: string;

  @SpeakeasyMetadata({ data: "json, name=startDateTime" })
  startDateTime?: string;

  @SpeakeasyMetadata({ data: "json, name=subject" })
  subject?: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: string;
}
